DFE-Digital/register-trainee-teachers-prototype-old
# Add concept of salary grants and make bursaries more generic

This PR adds support for collecting information about salary grants - which apply to the `School direct (salaried)` and `Early years (salaried)` routes.

Salary grants are basically the same as bursaries - they have a value `£` and they apply for certain subjects. Mostly what's different is the language.

To do this, I've made the existing bursary functionally more generic in name - it now calls it `financialSupport` and has a type.

## Testing:
* Add a trainee on a `School direct (salaried)` or `Early years (salaried)` route
* Add course details where the allocation subject ends up with one of those that has salary grants (all of early years or `chemistry, computing, maths, physics, languages, classics or biology`)
* When in funding section, the user is now asked about salary grants and is shown the correct value.

Salary grant page (school direct)
![Screenshot 2021-09-03 at 14 35 30](https://user-images.githubusercontent.com/2204224/132013937-6b90c19d-ef17-489b-a824-213666916535.png)

Summary card
![Screenshot 2021-09-03 at 14 35 39](https://user-images.githubusercontent.com/2204224/132013958-3f80fd21-196d-4fa8-a598-b9105f07b8cb.png)

Salary grant page (early years salaried)
![Screenshot 2021-09-03 at 14 37 45](https://user-images.githubusercontent.com/2204224/132014251-e523964f-2a35-4b12-ab6a-1994ebf516f3.png)
